### PR TITLE
Support non-standard AWS regions

### PR DESCRIPTION
**Pull request by viren-nadkarni**
Introduces a new config option `ALLOW_NONSTANDARD_REGIONS` which disables strict validation of AWS regions.

Closes https://github.com/localstack/localstack/issues/7993

### CODE REVIEW

File: tests/integration/test_stores.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import unittest
+
+
+@unittest.mock.patch("localstack.config.ALLOW_NONSTANDARD_REGIONS", True)
+def test_nonstandard_regions(create_boto_client, monkeypatch):
+    """
+    Ensure that non-standard AWS regions can be used vertically.
+    """
+    monkeypatch.setenv("MOTO_ALLOW_NONEXISTENT_REGION", "true")
```
Comment (viren-nadkarni) [inline tests/integration/test_stores.py:9]: Ideally we want to patch the config option before LS starts. What's a proper way to do this?

File: tests/integration/test_stores.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+    # Create a resource in Moto backend
+    ec2_client = create_boto_client("ec2", region_name="uranus-south-1")
+    ec2_client.create_key_pair(KeyName="foo")
+
+    # Create a resource in LocalStack store
+    sqs_client = create_boto_client("sqs", region_name="pluto-central-2a")
```
Comment (thrau) [inline tests/integration/test_stores.py:16]: :D

File: tests/integration/test_stores.py (hunk 1 above)
Comment (thrau) [reply tests/integration/test_stores.py:9]: unfortunately there is no good way of doing as at all. especially not in a test. config.py seems good for the runtime

**Review by thrau — APPROVED** [review]
nice addition!

**Comment by toadzky** [comment]
thanks guys

File: tests/integration/test_stores.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import unittest
+
+
+@unittest.mock.patch("localstack.config.ALLOW_NONSTANDARD_REGIONS", True)
```
Comment (dominikschubert) [inline tests/integration/test_stores.py:4]: @viren-nadkarni 

we usually use `monkeypatch.setattr(config, "ALLOW_NONSTANDARD_REGIONS", True)` here.
Any reason you're using unittest in this case?

Comment (viren-nadkarni) [reply tests/integration/test_stores.py:4]: No particular reason at all, it just didn't cross my mind. I'll fix it when I retouch this next time

**Pull request merged**